src/main/java/org/truffleruby/language/supercall/ReadZSuperArgumentsNode.java

```diff
@@ -48,7 +48,15 @@ public final Object execute(VirtualFrame frame) {
 
         if (restArgIndex != -1) {
             final Object restArg = superArguments[restArgIndex];
-            final Object[] restArgs = unsplat((RubyArray) restArg);
+
+            final Object[] restArgs;
+
+            if (restArg instanceof RubyArray) {
```

Comment:
Should we profile this condition (with ConditionProfile)?

Reply:
Could do - but I'll be doing a check of the performance of `super` after the specs pass so I'll update it then.

Reply:
I'd rather do it eagerly if you don't mind.
I think in general we currently profile every condition unless it folds at PE time or known to be worthless to profile.
I can also do it when integrating the PR.

Reply:
Sure you can do it then - there's a couple of unprofiled branches in this method then.

Reply:
Which other branches? `restArgIndex` is PE constant.